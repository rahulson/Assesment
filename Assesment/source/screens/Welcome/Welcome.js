import React, { useState } from 'react'
import { View, StyleSheet, Text } from 'react-native'
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { FormTextInput, FormButton, FormText } from '../../components'
import { SCREEN_CONSTANTS } from '../../constants/AppConstants'
import NavigationService from '../../navigation/NavigationService'
import NotifictaionMessage from '../../constants/NotificationMessage'
import { ColorConstants } from '../../constants/ColorConstants'
import FontSize from '../../constants/FontSize'
import isEmpty from 'lodash/isEmpty'
import { useAppContext } from '../../provider/UserProvider'
import { STORE_USER_NAME } from '../../store/UserAction'
import { isEmulator } from '../../utils/helper'

const Welcome = () => {

    const appContext = useAppContext()

    const dispatch = appContext && appContext.dispatch

    const [userName, setUserName] = useState('')

    function onTapEnter() {
        if (isEmpty(userName)) {
            NotifictaionMessage.showSimpleMessage('warning', { message: 'Please enter username', description: '' })
            return
        }
        dispatch({ type: STORE_USER_NAME, payload: userName })
        NavigationService.push(SCREEN_CONSTANTS.USERINFO)
    }

    function isSimulator() {
        return isEmulator() === 1 ? 'True' : 'False'
    }

    function renderDeviceCheck() {
        return (
            <View style={styles.contentContainer}>
                <FormText textStyle={styles.content}>Is App Running on Simulator?</FormText>
                <FormText textStyle={styles.content}>{isSimulator()}</FormText>
            </View>
        )
    }

    return (
        <View style={styles.container}>
            <Text style={styles.title}>Welcome</Text>
            {renderDeviceCheck()}
            <View style={styles.form}>
                <FormTextInput
                    containerStyle={styles.textContainer}
                    placeholder="Enter User Name"
                    autoCapitalize='none'
                    value={userName}
                    onChangeText={(value) => setUserName(value)}
                />
                <FormButton title={'Enter'} onPress={() => onTapEnter()} />
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: ColorConstants.WHITE,
        justifyContent: 'center',
        alignItems: 'center'
    },
    title: {
        alignSelf: 'center',
        textAlign: 'center',
        color: ColorConstants.PAINT_CHIP,
        fontSize: FontSize.extraLarge,
        fontWeight: '700',
        position: 'absolute',
        top: hp(8)
    },
    content: {
        color: ColorConstants.PAINT_CHIP, 
        fontSize: FontSize.tow, 
        fontWeight: '700', 
        marginBottom: hp(1)
    },
    contentContainer: {
        width: '100%', 
        marginBottom: hp(2), 
        justifyContent:'center', 
        alignItems:'center'
    },
    form: {
        width: '100%',
        paddingHorizontal: wp(10)
    },
    textContainer: {
        height: hp(6),
        width: '100%',
        backgroundColor: ColorConstants.WHITE,
        borderRadius: hp(1),
        marginBottom: hp(2)
    },
})

export default Welcome