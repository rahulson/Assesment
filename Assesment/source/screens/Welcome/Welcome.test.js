import React from 'react'
import renderer from 'react-test-renderer'

import Welcome from './Welcome'

it ("renders when there is no username entered", () => {
    const tree = renderer.create(<Welcome/>).toJSON()
    expect(tree).toMatchSnapshot()
})