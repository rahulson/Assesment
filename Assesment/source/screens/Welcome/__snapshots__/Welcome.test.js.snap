// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is no username entered 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#2F74DC",
        "fontSize": 66.5,
        "fontWeight": "700",
        "position": "absolute",
        "textAlign": "center",
        "top": 106.5,
      }
    }
  >
    Welcome
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 26.5,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2F74DC",
          "fontSize": 26.5,
          "fontWeight": "700",
          "marginBottom": 13.5,
        }
      }
      text=""
    >
      Is App Running on Simulator?
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#2F74DC",
            "fontSize": 26.5,
            "fontWeight": "700",
            "marginBottom": 13.5,
          },
          Object {
            "color": "green",
          },
        ]
      }
      text=""
    >
      False
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 75,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": "5%",
            "width": "100%",
          },
          Object {
            "backgroundColor": "white",
            "borderRadius": 13.5,
            "height": 80,
            "marginBottom": 26.5,
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Enter User Name"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "#c8c8c8",
              "borderRadius": 13.5,
              "borderWidth": 1,
              "height": 80,
              "paddingHorizontal": 13.5,
              "width": "100%",
            },
            Object {},
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2F74DC",
          "borderRadius": 13.5,
          "height": 80,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 26.5,
              "fontWeight": "600",
            },
            Object {},
          ]
        }
      >
        Enter
      </Text>
    </View>
  </View>
</View>
`;
